lib/Driver/ToolChains.cpp

```diff
@@ -52,6 +52,12 @@ bool ToolChain::JobContext::shouldUsePrimaryInputFileList() const {
   return InputActions.size() > TOO_MANY_FILES;
 }
 
+bool ToolChain::JobContext::shouldFilterInputsByType() const {
+  // SingleCompile's must not filter inputs by type in order for the
+  // playground (test) to work.
+  return OI.CompilerMode != OutputInfo::Mode::SingleCompile;
```

Comment:
If you're going to put in a hack for playgrounds, please make it dependent on the `-playground` flag, not WMO-or-not. It'd be better if we could change the other side, of course.

Reply:
This hack merely restores what the code did before. See calls to `types::isPartOfSwiftCompilation` in the relevant `constructInvocation` method. Is the comment misleading? What would be better?